### PR TITLE
Fix request method typescript

### PR DESCRIPTION
**Pull request by bkaney**
* Update TS for `client.request`
* Add ability to set bearerToken in the constructor
* Allow longer FHIR IDs that the spec (to support Epic)

### CODE REVIEW

**Comment by Dewar0019** [comment]
Should we update the readme and include the bearertoken example?

**Comment by bkaney** [comment]
> Should we update the readme and include the bearertoken example?

I could, it's optional but might be nice to show you can set it that way.

File: lib/client.js [hunk 1]
```diff
@@ -28,7 +28,8 @@ class Client {
    *     cert: certFileContent,
    *     key: keyFileContent,
    *     ca: caFileContent
-   *   }
+   *   },
+   *   bearerToken: 'eyJhbGci...dQssw5c'
```
Comment (johns397) [inline lib/client.js:32]: What is this default Bearer token value?  Is this for testing>

File: lib/client.js [hunk 2]
```diff
@@ -299,13 +305,17 @@ class Client {
    *   .then(data => console.log(data));
    *
    * @param {String} requestUrl - URL, can be relative to base or absolute
-   * @param {Object} paerams - (optional) Request params
+   * @param {Object} params - (optional) Request params
```
Comment (johns397) [inline lib/client.js:308]: Thanks for fixing this typo!

File: lib/fhir-regular-expressions.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
...
...l|MedicinalProductUndesirableEffect|MessageDefinition|MessageHeader|ModuleDefinition|ModuleMetadata|MolecularSequence|NamingSystem|NutritionOrder|NutritionRequest|Observation|ObservationDefinition|OccupationalData|OperationDefinition|OperationOutcome|Order|OrderResponse|OrderSet|Organization|OrganizationAffiliation|OrganizationRole|Patient|PaymentNotice|PaymentReconciliation|Person|PlanDefinition|Practitioner|PractitionerRole|Procedure|ProcedureRequest|ProcessRequest|ProcessResponse|ProductPlan|Protocol|Provenance|Questionnaire|QuestionnaireResponse|ReferralRequest|RelatedPerson|RequestGroup|ResearchDefinition|ResearchElementDefinition|ResearchStudy|ResearchSubject|RiskAssessment|RiskEvidenceSynthesis|Schedule|SearchParameter|Sequence|ServiceDefinition|ServiceRequest|Slot|Specimen|SpecimenDefinition|StructureDefinition|StructureMap|Subscription|Substance|SubstanceNucleicAcid|SubstancePolymer|SubstanceProtein|SubstanceReferenceInformation|SubstanceSourceMaterial|SubstanceSpecification|SupplyDelivery|SupplyRequest|Task|TerminologyCapabilities|TestReport|TestScript|UserSession|ValueSet|VerificationResult|VisionPrescription)\/[A-Za-z0-9\-.]{1,256}(\/_history\/[A-Za-z0-9\-.]{1,256})?$/,
```
Comment (johns397) [inline lib/fhir-regular-expressions.js:3]: Changing from max length = 64 (FHIR standard) to 256 (for non-compliant Epic) FHIR IDs makes sense.

**Review by johns397 — APPROVED** [review]
For the things I commented on.
I'm not sure of some of the other TypeScript code changes and the reasoning behind adding the RequestInitWithoutMethod.

File: lib/client.js (hunk 1 above)
Comment (bkaney) [reply lib/client.js:32]: This is part of the JSDoc

**Review by gilhsu — APPROVED** [review]

**Review by vstrateman — APPROVED** [review]
Looks good, but curious why you have to edit that type from node-fetch?

**Comment by bkaney** [comment]
> Looks good, but curious why you have to edit that type from node-fetch?

The signature of `client.request()` has `method` as a parameter. However `method` is also part of `RequestInit` (https://github.com/node-fetch/node-fetch/blob/main/%40types/index.d.ts#L65).

To avoid confusion, I wanted to document that you could not set `method` in the `options` parameter.

(FWIW we end up merging `method` in the requestBuilder, so it would always win if options.method was set -https://github.com/Vermonster/fhir-kit-client/blob/master/lib/http-client.js#L134-L139)

**Pull request merged**